pkg/messages/baseMessage.go

```diff
@@ -55,6 +57,12 @@ type RequestMessage interface {
 	Reply(senderFloor int, response bool) ResponseMessage
 }
 
+type ProposalMessage interface {
+	Message
+	Treaty() Treaty
+	Reply(senderFloor int, response bool) ResponseMessage
+}
+
 type ResponseMessage interface {
 	Message
```

Comment:
```suggestion
	Message
	Treaty() Treaty
```

Just for when baseAgent sticks this stuff into the memory, I'm pretty sure we'll have to extract the treaty from the response

Reply:
I disagree with adding `Treaty` to the response message... but then this has brought up the question of: how does the recipient agent know what the responseMessage is responding to?

Reply:
Nah I think I'll need it for the memory pushing - I need to handle a response and extract the treaty from there. I don't want to find the message this replies to

Reply:
I understand that, but multiple message types use `ResponseMessage` as a reply, not just `ProposalMessage`. Therefore, it shouldn't have `Treaty`.

Reply:
Then we need a new dataType to handle this

Reply:
Proposal message will need to have the reply of type TreatyResponseMessage that contains the original treaty

Reply:
I'd disagree with that - it would, I think, be better for agents to keep track of what messages they've sent (and therefore what treaties they've proposed). This allows `ResponseMessage()` to stay general.

Reply:
How about containing a pointer to the original message inside of here